File: survival-model/src/main/kotlin/dev/abarmin/survival/demo/scene/PixelColor.kt
package dev.abarmin.survival.demo.scene

/**
 * Color of one single piece of scene.
 *
 * @author Aleksandr Barmin
 */
data class PixelColor(
    val red: Int,
    val green: Int,
    var blue: Int,
    var transparency: Int
) {
    companion object {
        val TRANSPARENT = PixelColor(255, 255, 255, 100)
    }

    fun isTransparent(): Boolean {
        return transparency == 100
    }
}

File: survival-world-builder-web/src/app/pixel-color.ts
export type PixelColor = {
    red: number
    green: number
    blue: number
    transparency: number
}

export function colorCopy(color: PixelColor): PixelColor {
    return {
        red: color.red,
        green: color.green,
        blue: color.blue,
        transparency: color.transparency
    }
}